Extension/ui/settings.ts

```diff
@@ -212,10 +212,24 @@ class SettingsApp {
         this.vsCodeApi.postMessage({
             command: "knownCompilerSelect"
         });
+    }
 
-        // Reset selection to none
-        el.value = "";
+    // To enable custom entries, the compiler path control is a text box on top of a select control.
+    // This function ensures that the select control is updated when the text box is changed.
+    private fixKnownCompilerSelection(): void {
+        const compilerPath = <HTMLInputElement>document.getElementById(elementId.compilerPath);
+        const knownCompilers = <HTMLSelectElement>document.getElementById(elementId.knownCompilers);
+        for (let n = 0; n < knownCompilers.options.length; n++) {

```

Comment:
Can you do 

```typescript
        for (let n = 0; n < knownCompilers.options.length; n++) {
            if (compilerPath.value.toLowerCase() === knownCompilers.options[n].value.toLowerCase()) {
                knownCompilers.value = knownCompilers.options[n].value;
                return;
            }
            knownCompilers.value = '';
        }
````

Reply:
I refactored this into a function after some testing revealed I needed to do it in two places, but I didn't take advantage of the fact that I could do an early return now.

I don't think I want to set the value to `''` N times during this loop though.  What I should do is move that out of the loop and let the early return handle the other case.